src/notebooklm/_retry.py

```
@@ -0,0 +1,335 @@
+                    attempt,
+                    max_attempts,
+                    delay,
+                )
+                await asyncio.sleep(delay)
+                # Apply backoff AFTER sleeping for the next iteration
+                if isinstance(e, RateLimitError) and e.retry_after is not None:
+                    delay = float(e.retry_after)
+                else:
+                    delay *= backoff_factor
+                    if jitter:
+                        delay *= 0.75 + random.random() * 0.5
```

Ruff (0.15.11)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)